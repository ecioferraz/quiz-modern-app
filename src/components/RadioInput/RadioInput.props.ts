import { ReactNode } from 'react';

type RadioInputProps = { children: ReactNode };

export default RadioInputProps;
