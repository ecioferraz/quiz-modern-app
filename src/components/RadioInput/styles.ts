import styled, { css } from 'styled-components';
import RadioInputProps from './RadioInput.props';

export const Container = styled.section<{ answer: RadioInputProps['answer'] }>`
  ${({ answer, theme }) => css`
    align-items: center;
    background: ${answer === 'wrong' && '#F6E5E5'};
    border: ${answer === 'correct' && '1px solid #219653'};
    border-radius: 10px;
    box-shadow: 10px 24px 54px rgba(51, 51, 51, 0.05);
    display: flex;
    gap: ${theme.spacings.small};
    margin-top: ${theme.spacings.small};
    padding: ${theme.spacings.small};

    img {
      width: ${theme.spacings.medium};
    }

    p {
      font-size: 1.4rem;
      font-weight: ${theme.font.weight.medium};
      margin: 0;
    }
  `}
`;
