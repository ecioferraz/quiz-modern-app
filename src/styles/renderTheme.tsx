import { render, RenderResult } from '@testing-library/react';
import { ReactNode } from 'react';
import { ThemeProvider } from 'styled-components';
import { theme } from './theme';

export default function renderTheme(children: ReactNode): RenderResult {
  return render(<ThemeProvider theme={theme}>{children}</ThemeProvider>);
}
