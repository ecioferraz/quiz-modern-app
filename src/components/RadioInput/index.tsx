import SvgCard from '../SvgCard';
import TextCard from '../TextCard';
import RadioInputProps from './RadioInput.props';
import * as Styled from './styles';

export default function RadioInput({ answer, iconSrc, option }: RadioInputProps) {
  return (
    <Styled.Container answer={answer}>
      {/* functionally could be something like this */}
      {/* {options.map((option) => (
        <label htmlFor={option} key={option}>
          <input id={option} name="question" type="radio" value={option} />
          {option}
        </label>
      ))} */}
      <SvgCard alt="Radio option" src={iconSrc} />
      <TextCard as="p">{option}</TextCard>
    </Styled.Container>
  );
}
