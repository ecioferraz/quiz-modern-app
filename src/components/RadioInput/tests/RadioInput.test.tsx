import renderTheme from '../../../styles/renderTheme';
import RadioInput from '..';
import { describe, it } from 'vitest';

describe('<RadioInput />', () => {
  it('should render <RadioInput />', () => {
    const { container } = renderTheme(
      <RadioInput iconSrc="assets/icons/circle.svg" option="option test" />,
    );

    expect(container).toMatchSnapshot();
  });

  it('should render <RadioInput /> with correct answer', () => {
    const { container } = renderTheme(
      <RadioInput
        answer="correct"
        iconSrc="assets/icons/circle.svg"
        option="option test"
      />,
    );

    expect(container).toMatchSnapshot();
  });

  it('should render <RadioInput /> with wrong answer', () => {
    const { container } = renderTheme(
      <RadioInput
        answer="wrong"
        iconSrc="assets/icons/circle.svg"
        option="option test"
      />,
    );

    expect(container).toMatchSnapshot();
  });
});
