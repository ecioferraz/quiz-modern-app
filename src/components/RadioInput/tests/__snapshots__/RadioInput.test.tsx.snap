// Vitest Snapshot v1

exports[`<RadioInput /> > should render <RadioInput /> 1`] = `
.c2 {
  font-size: 1.6rem;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 10px;
  box-shadow: 10px 24px 54px rgba(51,51,51,0.05);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 1.6rem;
  margin-top: 1.6rem;
  padding: 1.6rem;
}

.c0 img {
  width: 2.4rem;
}

.c0 p {
  font-size: 1.4rem;
  font-weight: 500;
  margin: 0;
}

<div>
  <section
    class="c0"
  >
    <section
      class="c1"
    >
      <img
        alt="Radio option"
        src="assets/icons/circle.svg"
      />
    </section>
    <p
      class="c2"
    >
      option test
    </p>
  </section>
</div>
`;

exports[`<RadioInput /> > should render <RadioInput /> with correct answer 1`] = `
.c2 {
  font-size: 1.6rem;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: 1px solid #219653;
  border-radius: 10px;
  box-shadow: 10px 24px 54px rgba(51,51,51,0.05);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 1.6rem;
  margin-top: 1.6rem;
  padding: 1.6rem;
}

.c0 img {
  width: 2.4rem;
}

.c0 p {
  font-size: 1.4rem;
  font-weight: 500;
  margin: 0;
}

<div>
  <section
    class="c0"
  >
    <section
      class="c1"
    >
      <img
        alt="Radio option"
        src="assets/icons/circle.svg"
      />
    </section>
    <p
      class="c2"
    >
      option test
    </p>
  </section>
</div>
`;

exports[`<RadioInput /> > should render <RadioInput /> with wrong answer 1`] = `
.c2 {
  font-size: 1.6rem;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: #F6E5E5;
  border-radius: 10px;
  box-shadow: 10px 24px 54px rgba(51,51,51,0.05);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 1.6rem;
  margin-top: 1.6rem;
  padding: 1.6rem;
}

.c0 img {
  width: 2.4rem;
}

.c0 p {
  font-size: 1.4rem;
  font-weight: 500;
  margin: 0;
}

<div>
  <section
    class="c0"
  >
    <section
      class="c1"
    >
      <img
        alt="Radio option"
        src="assets/icons/circle.svg"
      />
    </section>
    <p
      class="c2"
    >
      option test
    </p>
  </section>
</div>
`;
